### PR TITLE
Feature/proptypes

### LINKED ISSUE #25: proptypes!!

### PR DESCRIPTION
**Pull request by espressoGoddess**
### Files Changed:
- `package-lock.json`, `package.json`, `Activities.js`, `FeelingsLog.js`, `Feelings.js`, `JournalEntry.js`, `JournalPrompt.js`

### What Has Changed: 
- This installs the proptypes dependency
- This adds proptypes to all components with props being received. This specifies the type within each object and array.

### Notes: 
- Do we want to add default proptype values? I did not add them because I wasn't sure what to put for it.

### Closes:
Closes #25

### CODE REVIEW

File: client/src/Components/Activities/Activities.js [hunk 1]
```diff
@@ -36,4 +37,14 @@ const Activities = ({ activities, addEntry, setFeeling }) => {
   )
 }
 
-export default Activities;
\ No newline at end of file
+export default Activities;
+
+Activities.propTypes = {
+  activities: PropTypes.arrayOf(PropTypes.shape({
```
Comment (jfogiato) [inline client/src/Components/Activities/Activities.js:43]: Oooo the `.shape()` is a new one for me! Great call implementing this. Definitely want to keep this method in mind when using this package in the future.

File: client/src/Components/FeelingsLog/FeelingsLog.js [hunk 2]
```diff
@@ -39,9 +39,21 @@ const FeelingsLog = ({logs}) => {
...
+FeelingsLog.propTypes = {
+  logs: PropTypes.arrayOf(PropTypes.shape({
+    entry_id: PropTypes.number.isRequired,
+    user_id: PropTypes.string,
+    date: PropTypes.string.isRequired,
+    feeling: PropTypes.string.isRequired,
+    journal_entry: PropTypes.string.isRequired,
+    created_at: PropTypes.string,
+    updated_at: PropTypes.string,
+    activity: PropTypes.string.isRequired
```
Comment (jfogiato) [inline client/src/Components/FeelingsLog/FeelingsLog.js:57]: Another awesome use of `.shape()`!! I love this formatting, too - it makes it really easy to look at this and understand exactly what you're expecting to be passed in. Awesome work!!

File: client/src/Components/JournalEntry/JournalEntry.js [hunk 3]
```diff
@@ -24,4 +24,17 @@ const JournalEntry = ({ journal }) => {
...
+JournalEntry.propTypes = {
+  journal: PropTypes.shape({
+    date: PropTypes.string.isRequired,
+    feeling: PropTypes.string.isRequired,
+    activity: PropTypes.string.isRequired,
+    journal_entry: PropTypes.string.isRequired,
+    entry_id: PropTypes.number,
+    created_at: PropTypes.string,
+    updated_at: PropTypes.string,
+    user_id: PropTypes.string,
```
Comment (jfogiato) [inline client/src/Components/JournalEntry/JournalEntry.js:38]: Same thing here!

**Review by jfogiato — APPROVED** [review]
Looks great!! Thanks for doing all this work and being so thoughtful in the organization and types you checked for!!

File: client/src/Components/Activities/Activities.js (hunk 1 above)
Comment (KatherineBlaine) [reply client/src/Components/Activities/Activities.js:43]: Shape is a new one for me too! This looks great, super thorough.

**Review by KatherineBlaine — APPROVED** [review]
This is great, Amber. Super thorough and adds a lot to the integrity of the application!

File: client/src/Components/JournalPrompt/JournalPrompt.js [hunk 4]
```diff
@@ -16,4 +17,9 @@ const JournalPrompt = ( {updateJournal, feeling} ) => {
   )
 }
 
-export default JournalPrompt;
\ No newline at end of file
+export default JournalPrompt;
+
+JournalPrompt.propTypes = {
+  updateJournal: PropTypes.func.isRequired,
+  feeling: PropTypes.string.isRequired
```
Comment (chrissycooper) [inline client/src/Components/JournalPrompt/JournalPrompt.js:24]: Super nice work Amber! No notes!!!! Thank you for taking this on

**Review by chrissycooper — APPROVED** [review]

**Pull request merged**